src/main/java/core/basesyntax/strategy/SupplyOperationHandler.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.serviceimpl.FruitTransactionValidatorImpl;
+import java.util.Optional;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private FruitTransactionValidatorImpl validator;
+
+    @Override
```

Comment:
The `apply` method should not be void. It should return the updated storage or the result of the operation. This would make the method more testable and adhere to the single responsibility principle.